packages/schema/src/Schema.ts

```diff
@@ -1889,7 +1908,7 @@ export const optional: {
       | (Types.Has<Options, "as"> extends true ? option_.Option<A> : A)
       | (Types.Has<Options, "as" | "default" | "exact"> extends true ? never : undefined),
       never,
-      "?:",
+      Types.Has<Options, "onNoneEncoding"> extends true ? ":" : "?:",
```

Comment:
This doesn't seem correct, it should always be `"?:"` since you are declaring an optional property

Reply:
I understand. I wanted to type properly the output of the encode but ot seems like I can't do that without impacting the input type of `S.decode(schema)`, right?
Is it ok to keep `"?:"` even though we know that the encode output will always be present? Or you think it's a bad idea to add this `onNoneEncoding`?

Reply:
> I can't do that without affecting the input type of `S.decode(schema)`, correct?

right

> Is it okay to keep `"?:"` even though we know that the encoded output will always be present?

yes because the intention here is to define an optional property within the `I` type of the schema